tests/pass/pass_test.go

```diff
@@ -0,0 +1,192 @@
+	"testing"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+type ConfigureBrowserExtensionResponse struct {
+	BrowserExtensionPairingToken string `json:"browser_extension_pairing_token"`
+	ConnectionToken              string `json:"connection_token"`
+}
+
+var (
+	httpClient = http.DefaultClient
+	wsDialer   = websocket.DefaultDialer
+)
+
+func TestPassHappyFlow(t *testing.T) {
+	const api = "localhost:8082"
+
+	resp, err := configureBrowserExtension("http://" + api + "/browser_extension/configure")
+	if err != nil {
+		t.Fatalf("Failed to configure browser extension: %v", err)
+	}
+
+	browserExtensionDone := make(chan struct{})
+	mobileDone := make(chan struct{})
+
+	go func() {
+		defer close(browserExtensionDone)
+
+		err := browserExtensionWaitForConfirm("ws://"+api+"/browser_extension/wait_for_connection", resp.BrowserExtensionPairingToken)
+		if err != nil {
+			t.Errorf("Error when Browser Extension waited for confirm: %v", err)
+			return
+		}
+
+		err = proxyWebSocket(
+			"ws://"+api+"/browser_extension/proxy_to_mobile",
+			resp.BrowserExtensionPairingToken,
+			"sent from browser extension",
+			"sent from mobile")
+		if err != nil {
+			t.Errorf("Browser Extension: proxy failed: %v", err)
+			return
+		}
+
+	}()
+	go func() {
+		defer close(mobileDone)
+
+		err := confirmMobile("http://"+api+"/mobile/confirm", resp.ConnectionToken)
+		if err != nil {
+			t.Errorf("Mobile: confirm failed: %v", err)
+			return
+		}
+
+		err = proxyWebSocket(
+			"ws://"+api+"/mobile/proxy_to_browser_extension",
+			resp.BrowserExtensionPairingToken,
+			"sent from mobile",
+			"sent from browser extension",
+		)
+		if err != nil {
+			t.Errorf("Mobile: proxy failed: %v", err)
+			return
+		}
+	}()
+
+	<-browserExtensionDone
+	<-mobileDone
+}
+
+func browserExtensionWaitForConfirm(url, token string) error {
+	var resp struct {
+		Status string `json:"status"`
+	}
+
+	conn, err := dialWS(url, token)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	conn.SetReadDeadline(time.Now().Add(time.Second))
+	_, message, err := conn.ReadMessage()
+	if err != nil {
+		return fmt.Errorf("error reading from connection: %w", err)
+	}
+	if err := json.Unmarshal(message, &resp); err != nil {
+		return fmt.Errorf("failed to decode message: %w", err)
+	}
+	const expectedStatus = "ok"
+	if resp.Status != expectedStatus {
+		return fmt.Errorf("received status %q, expected %q", resp.Status, expectedStatus)
+	}
+	return nil
+}
+
+func configureBrowserExtension(endpoint string) (ConfigureBrowserExtensionResponse, error) {
+	req, err := http.NewRequest("POST", endpoint, bytes.NewBufferString(`{"extension_id":"9317f6c1-0470-41c3-b735-542f2c6b70c3"}`))
+	if err != nil {
+		return ConfigureBrowserExtensionResponse{}, fmt.Errorf("failed to create http request: %w", err)
+	}
+	httpResp, err := httpClient.Do(req)
+	if err != nil {
+		return ConfigureBrowserExtensionResponse{}, fmt.Errorf("failed perform the request: %w", err)
+	}
+	defer httpResp.Body.Close()
+
```

Comment:
i guess you should check for response here and if it's >=400 you should try to unmarshal it to map, or fail